internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1139,6 +1185,50 @@ func TestMachineSetReconciler_syncMachines(t *testing.T) {
 		HaveValue(Equal(*ms.Spec.Template.Spec.NodeDrainTimeout)),
 	))
 
+	// The in-place mutating infrastructure machine should have:
+	// - clean-up managed fields
+	// - updated in-place propagating values
+	updatedInfraMachine := infraMachine.DeepCopy()
+	g.Expect(env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(updatedInfraMachine), updatedInfraMachine))
+
+	// Verify ManagedFields
+	g.Expect(updatedInfraMachine.GetManagedFields()).Should(
```

Comment:
Let's extend this validation once we changed the prod code to verify that the ssa manager owns labels/annotations and "manager" some other spec field that we added

prior art: https://github.com/kubernetes-sigs/cluster-api/blob/4d3b1817e3ee53e34cef21575bb29ec2575b4744/internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go#L696

Let's also check that the spec field stayed the same

(same for bootstrapconfig of course)